test/functional/Valhalla/src/org/openj9/test/lworld/ValueTypeUnsafeTests.java

```diff
@@ -463,4 +508,253 @@ static public void testGetSizeOfZeroSizeVT() {
 			assertEquals(size, 8);
 		}
 	}
+
+	@Test(dataProvider = "compareAndDoSomethingFuncs")
+	static public void testCompareAndSetNullObj(CompareAndDoSomethingFunction compareAndSwapValue) throws Throwable {
+		if (isFlatteningEnabled) {
```

Comment:
I guess it segfaults in the `isFlatteningEnabled == false` case here ?

Reply:
That's probably what I thought when I was writing the tests but it turns out it's just an NPE
I changed it to also test the unflattened case in cf98cd3